Add timing details for userSeat promise returns (#834)
Agoric/documentation: main/reference/zoe-api/user-seat.md

Take care when sharing a **UserSeat**, since it includes authority to (attempt to) exit the seat.

These methods are all documented as returning Promises, since they are always called remotely,
but that obscures the fact that they are expected to resolve at semantically different times.

> Some return a value immediately, some do a small amount of computation before returning, and

Reviewer: ```suggestion
Some return a value promptly, some do a small amount of computation before returning, and
```

You use "promptly" below, and for this purpose it is much clearer than immediately. Do we define either term anywhere? I like Norm's definition of "prompt" or "promptly", that it will happen without any further actions needed by any other party. Also, I would say, implies only computational delays such as round trips, rather than scheduled timing delays.

Thus, unclear if you meant "do a small amount of computation before returning" to be a separate category. From the pov of a remote caller, I'd expect these to also be prompt and to not be in a different category from your "immediately".

I would like to reserve terms like "immediate" and "immediately" for things that happen synchronously in the same turn, which is never true for remote calls.

**Payment** associated with the *keyword* argument.

This promise will be resolved when the seat exits.

## E(UserSeat).getOfferResult()
  - Returns: **Promise&lt;OfferResult>**

Returns a **Promise** for an **OfferResult**. The **OfferResult** can be literally anything. 
For example, in tests for the Automated Refund Dapp, it's the string "The offer was accepted". In
the Covered Call example, it's a call option, which is an assayable **[Invitation](./zoe-data-types.md#invitation)**
to buy the underlying asset. Strings and invitations are the most common things returned.
The value is set by the returned result of the **offerHandlers** function passed
as an argument to **[zcf.makeInvitation()](./zoe-contract-facet.md#zcf-makeinvitation-offerhandler-description-customproperties-proposalshape)**.

Since the contract can return whatever it wants as an offerResult, there is no guarantee that the
promise will resolve promptly.

## E(UserSeat).hasExited()
  - Returns: **Promise&lt;Boolean>**

Returns **true** if the seat has exited, **false** if it's still active. The value is returned

> immediately.

Reviewer: ```suggestion
promptly.
```

If you want to take some action when the seat does exit, use **getExitSubscriber()**
and use
```
const subscriber = E(seat).getExitSubscriber();
E.when(E(subscriber).getUpdateSince(), () => takeAction());

> ```

Reviewer: The difference being that subscribing to a remote contract remains valid across upgrade of the contract, whereas a remote promise to be resolved by a contract will instead be broken by an intervening contract upgrade.

> await 

Reviewer: Dangling `await`?

| 0 | The user didn't get what they wanted from the **Proposal**, so their offer was refunded. |
| 1 | The user got what they wanted from the **Proposal**, so their offer is spent & gone. |

> This promise will be resolved when the seat exits.

Reviewer: ```suggestion
This promise will be resolved promptly once the seat exits.
```

Is technically more precise, but the original text is fine if this precision is unneeded.

- Returns: **Promise&lt;[Allocation](./zoe-data-types.md#allocation)>**

Returns a **Promise** for the **Allocation** when the **UserSeat** exits the **proposal**.

> This promise will be resolved when the seat exits.

Reviewer: Likewise
